Autodesk/synthesis
# Exporter APS Authentication

Changes:
- Palette to get authorization code for Autodesk account
- Fetches code and refreshes automatically once expired
- Shows first name in a new tab in the main exporter panel (temporary)

Testing:
- Make sure the server is running (same branch name as this pr)
- Sign in with APS button
- Check main panel to see if your name shows up in a third tab
- Make sure login palette automatically closes after signing in
- Make sure reopening palette prompts for login again
- Make sure closing palette manually and reopening prompts for login again
  - Might have to enter email but not password to ensure that it's resetting

> [!WARNING]
> I haven't yet tested refreshing the token once it's expired. The web request works though and gets a different access token than before.

> [!IMPORTANT]
> This branch must be tested while running the server through the Docker container, because some authorization logic happens on the server side.
